app/uk/gov/hmrc/leakdetection/services/ReportsService.scala

```diff
@@ -82,6 +86,27 @@ class ReportsService @Inject()(
     } yield ()
   }
 
+  def saveLeaks(report: Report):Future[Unit] = {
+    val leaks = report.inspectionResults.map(r => Leak(
+      repoName    = report.repoName,
+      branch      = report.branch,
+      timestamp   = report.timestamp,
+      reportId    = report.id,
+      ruleId      = r.ruleId.getOrElse("Unknown"),
+      description = r.description,
+      filePath    = r.filePath,
+      scope       = r.scope,
+      lineNumber  = r.lineNumber,
+      urlToSource = r.urlToSource,
+      lineText    = r.lineText,
+      matches     = r.matches,
+      priority    = r.priority,
+      isTruncated = r.isTruncated.getOrElse(false))
+    )
+    // todo: make use to the stats returned by the update, maybe send to timeline when its done?
+    leakRepository.update(report.repoName, report.branch, leaks).map(_ => ())
```

Comment:
There's nothing wrong with using the inspection results from the report and storing them all at the end but it will require refactoring when we do the clean up and get rid of the inspection results from the report.

Reply:
yeah my intention was to do it in steps, get the leaks writing to their own collection first, then once we're happy its works correctly, refactor the report object